l1-contracts/test/foundry/unit/concrete/Bridgehub/experimental_bridge.t.sol

```diff
@@ -24,24 +25,49 @@ contract ExperimentalBridgeTest is Test {
     address public bridgeOwner;
     DummyStateTransitionManagerWBH mockSTM;
     DummyHyperchain mockChainContract;
-    DummySharedBridge mockSharedBridge;
-    DummySharedBridge mockSecondSharedBridge;
+    L1SharedBridge sharedBridge;
+    address sharedBridgeAddress;
+    address secondBridgeAddress;
+    L1SharedBridge secondBridge;
     TestnetERC20Token testToken;
+    address mockL2Contract;
 
     uint256 eraChainId;
 
+    event NewChain(uint256 indexed chainId, address stateTransitionManager, address indexed chainGovernance);
+
     function setUp() public {
         eraChainId = 9;
         bridgeHub = new Bridgehub();
         bridgeOwner = makeAddr("BRIDGE_OWNER");
         mockSTM = new DummyStateTransitionManagerWBH(address(bridgeHub));
         mockChainContract = new DummyHyperchain(address(bridgeHub), eraChainId);
-        mockSharedBridge = new DummySharedBridge(keccak256("0xabc"));
-        mockSecondSharedBridge = new DummySharedBridge(keccak256("0xdef"));
+        mockL2Contract = makeAddr("mockL2Contract");
+
+        // mocks to use in bridges instead of using a dummy one
+        address mockL1WethAddress = makeAddr("Weth");
+        address eraDiamondProxy = makeAddr("eraDiamondProxy");
+
+        sharedBridge = new L1SharedBridge(mockL1WethAddress, bridgeHub, eraChainId, eraDiamondProxy);
+        address defaultOwner = sharedBridge.owner();
+        vm.prank(defaultOwner);
+        sharedBridge.transferOwnership(bridgeOwner);
+        vm.prank(bridgeOwner);
+        sharedBridge.acceptOwnership();
+
+        secondBridge = new L1SharedBridge(mockL1WethAddress, bridgeHub, eraChainId, eraDiamondProxy);
+        defaultOwner = secondBridge.owner();
+        vm.prank(defaultOwner);
+        secondBridge.transferOwnership(bridgeOwner);
+        vm.prank(bridgeOwner);
+        secondBridge.acceptOwnership();
+
+        sharedBridgeAddress = address(sharedBridge);
+        secondBridgeAddress = address(secondBridge);
         testToken = new TestnetERC20Token("ZKSTT", "ZkSync Test Token", 18);
```

Comment:
Can we aim for tests with tokens that have decimals other than 18, such as 6 (USDC) , cTokens from Compound and WBTC have 8, etc.

Reply:
Added tokens with different decimals, I've used modifier to choose random token.